### DISCUSSION TITLE
Error finding Lat, Lon values in soil parameter file for VIC 4.1.2 LSM and Unable to find porosity map file for Mosaic LSM

### DISCUSSION THREAD

**Discussion by Gayathridevi123** [NASA-LIS/LISF#991] [Q&A] [answered]
Hi all,

**1. VIC.4.1.2** 

I am trying to force the VIC.4.1.2 LSM using GDAS & using the LISF 7.3.1 version.  But getting error :-
***** VIC Version 4.1.2.l bug fix update 2014-Apr-03 *****
**ERR: Error finding (6.375000, 80.125000) in the flat ascii soil paramter file.
ERR: Error finding (6.375000, 80.375000) in the flat ascii soil paramter file.
ERR: Error finding (6.375000, 80.625000) in the flat ascii soil paramter file.
ERR: Error finding (6.375000, 80.875000) in the flat ascii soil paramter file.
.
.
.
ERR: Error finding (38.375000, 99.375000) in the flat ascii soil paramter file.
ERR: Error finding (38.375000, 99.625000) in the flat ascii soil paramter file.
ERR: Error finding (38.375000, 99.875000) in the flat ascii soil paramter file.
Error in vegetation file.  Grid cell 0 not found..**

 LDT/LIS domain and grid is 
Run domain lower left lat:          6.375
Run domain lower left lon:         66.375
Run domain upper right lat:        38.375
Run domain upper right lon:       99.875
Run domain resolution (dx):          0.25
Run domain resolution (dy):          0.25

The corresponding parameter files are given from  https://portal.nccs.nasa.gov/lisdata_pub/data/PARAMETERS/VIC_PARAMETERS/1deg/.
ldt.config, lis.config, error file are attached.

[ldt.config.txt](https://github.com/NASA-LIS/LISF/files/8108921/ldt.config.txt)
[error.txt](https://github.com/NASA-LIS/LISF/files/8108923/error.txt)
[lis.config.txt](https://github.com/NASA-LIS/LISF/files/8108924/lis.config.txt)

**2. Mosaic LSM**

Tried to force Mosaic LSM using GDAS. 
 [INFO] GDAS forcing directory : ./input/MET_FORCING/GDAS/
 MSG: The GDAS forcing resolution is coarser than the running domain.
    Interpolating with the bilinear method.
  setmosp -- reading soil files
 Reading soil parameter table ./input/PARAMETERS/mos_parms/mosaic_soilparms_fao.txt          
setmosp -- no porosity datasource selected
  setmosp -- calling endrun
 [ERR] endrun is being called..

**porosity map is needed for running mosaic and I'm unable to find this file from https://portal.nccs.nasa.gov/lisdata_pub/data/PARAMETERS/UMD/1KM/...**

Would you please let me know where to get the porosity map for running mosaic LSM?

Please kindly let me know how to solve these two issues. 

Thanks a lot!
Regards,
Gayathri

**Comment by dmocko** [comment, marked as answer]
Hi Gayathri,

About VIC.4.1.2, it looks like you are trying to run at 0.25-degree grid spacing, but using 1-km parameters.  You will need to use parameters at 0.25-degrees.  Unfortunately, we don't have parameters at this grid spacing.  Please see the VIC documentation for more about these parameter files.  You may be able to create these parameter files for your running grid:
    https://vic.readthedocs.io/en/master/Documentation/Drivers/Classic/SoilParam/

About Mosaic, the parameter files that you need are in this location on the LIS data portal:
    https://portal.nccs.nasa.gov/lisdata_pub/data/PARAMETERS/mos_parms/

David

**Reply by Gayathridevi123** [reply]
Hi,
Thank you for your response David@dmocko.  I'm trying with Mosaic LSM and when I set the 
    Porosity data source: none
    Porosity map: 
The model was giving error:-
      setmosp -- reading soil files
      Reading soil parameter table ./input/PARAMETERS/mos_parms/mosaic_soilparms_fao.txt               
      setmosp -- no porosity datasource selected
      setmosp -- calling endrun
      [ERR] endrun is being called
Then I have seen from https://github.com/NASA-LIS/LISF/blob/master/lis/surfacemodels/land/mosaic/setmosp.F90 line 263; 
if porosity is set as "none" then end-run is called. I'm not able to find the porosity file for 1km,5km,10km, or 25km so I kept the Porosity data source: CONSTANT, and there was no error related to porosity.

But now I'm getting error :-
**At line 129 of file ../surfacemodels/land/mosaic/mosdynp.F90 (unit = 16, file = './input/PARAMETERS/mos_parms/mosaic_monthlyvegparms_umd.txt')
Fortran runtime error: Bad real number in item 1 of list input**
Error termination. Backtrace:
#0  0x2b6e28a2aac8 in read_real
	at ../../../libgfortran/io/list_read.c:2026
#1  0x2b6e28a2cead in list_formatted_read_scalar
	at ../../../libgfortran/io/list_read.c:2180
#2  0xaeb587 in ???
#3  0x59cdeb in ???
#4  0x5972e0 in ???
#5  0x5b87d1 in ???
#6  0xd3cbe2 in ??? 
#7  0x5cbc84 in ???
#8  0x4084b3 in ???
#9  0x2b6e296213d4 in ???
#10  0x408568 in ???
#11  0xffffffffffffffff in ???

Given the Mosaic LSM parameters:-  (from https://portal.nccs.nasa.gov/lisdata_pub/data/PARAMETERS/mos_parms/)
        Mosaic vegetation parameter table:               ./input/PARAMETERS/mos_parms/mosaic_vegparms_umd.txt
        Mosaic monthly vegetation parameter table: ./input/PARAMETERS/mos_parms/mosaic_monthlyvegparms_umd.txt
        Mosaic soil parameter table:                          ./input/PARAMETERS/mos_parms/mosaic_soilparms_fao.txt

and mosaic monthly vegetation parameter table has values for 13 UMD vegtypes.  Can you please have a look into this issue?

Thank you,
Regards,
Gayathri

**Comment by Chanhyuk-Choi** [comment]
Hi, Gayathri,

I'm Choi, just studying LIS..
Are you success your VIC or Mosaic running? 

I also tried to VIC and Mosaic running but, have a lot of errors.. 
If you succeed in your running the variable environment, can I receive your VIC and Mosaic ldt, lis.config file..? 

Thank you.

**Reply by Gayathridevi123** [reply]
Hi Choi,

Yeah, I'm able to run Mosaic LSM successfully.  For VIC, some parameter files were not available in the resolution that I want, so not tried again with VIC. 

For Mosaic, a bug was there in the previous version of LISF and I'm not sure they updated that in LISF 7.3.2 version. In recent versions of LISF, I think they updated the bug for Mosaic. Just go through #994 this discussion regarding the bug and please find the attached ldt and lis config files for Mosaic. Try it and see if it works and let me know if you need any further help in this regard.

[ldt.txt](https://github.com/NASA-LIS/LISF/files/11155910/ldt.txt)
[lis.txt](https://github.com/NASA-LIS/LISF/files/11155911/lis.txt)

Thank you
Gayathri

**Comment by Chanhyuk-Choi** [comment]
Hi Gayathri,

Thank you for your kind reply.

I trying to run VIC continuously, If I can run, I will share my ldt, lis
config file for VIC model.

Thank you very much
Choi

**Comment by Chanhyuk-Choi** [comment]
Hello @Gayathridevi123,

I tried to Mosaic LSM but, I had an error for running with lis.config files.
ldt.config was completed successfully, but there seems to be an error in lis.config execution because of a problem with ldt output ncfile.

can you please check it once?

1. This image is an error message while running lis.config
![image](https://user-images.githubusercontent.com/73869049/234441054-4c5c9429-5539-4506-90f7-b52848155256.png)

2. ldt.config file is

[ldt.config.mosaic.0.5grid.gldas.txt](https://github.com/NASA-LIS/LISF/files/11327998/ldt.config.mosaic.0.5grid.gldas.txt)

3.  ldt.config output ncfile is

[lis_input.gldas_0.5grid.mosaic.d01.zip](https://github.com/NASA-LIS/LISF/files/11328005/lis_input.gldas_0.5grid.mosaic.d01.zip)

4. lis.config file is

[lis.config.era5.mosaic.txt](https://github.com/NASA-LIS/LISF/files/11328008/lis.config.era5.mosaic.txt)

**Reply by Gayathridevi123** [reply]
Hi @Choi-chanhyuck 

Can you attach the log files?

**Reply by Chanhyuk-Choi** [reply]
Here is the log file.

Thank you very much @Gayathridevi123 

[lislog.00.0000.txt](https://github.com/NASA-LIS/LISF/files/11328994/lislog.00.0000.txt)

**Reply by Gayathridevi123** [reply]
In your 20 log files, any log file is showing an error message? I mean showing the end run. If so can you share that log file? Sorry, till next week I'm not able to give it a try.

**Reply by Chanhyuk-Choi** [reply]
My processor setting seems to use 20 processors, but since I execute only one processor so, log file Is only one.
If I try with a 1-process setup, I get the same error. 
Thank you.

**Reply by Gayathridevi123** [reply]
Hi @Choi-chanhyuck,

Which LIS version you are using? when I was trying Mosaic there was a bug and I guess it is resolved in the new version of LIS only. https://github.com/NASA-LIS/LISF/discussions/994 This was a discussion regarding that bug..And your error seems similar to this issue

**Reply by Chanhyuk-Choi** [reply]
Hi @Gayathridevi123 
My version is 7.4.0.. after seeing your discussion so I upgraded the version to 7.4.0, 
I checked the script to fix the bug, and the version I'm using has a script fixed..

**Comment by dmocko** [comment]
Hi @Choi-chanhyuck 

Are you sure the error message you posted (line 153 of umd_sibalb.F90) is the same run as the lislog.00.0000.txt file you provided?

I ask because the end of the lislog file indicates the problem is in the GLDAS forcing.  But I suppose it's possible that bad forcing data you are providing to Mosaic is what is causing the crash.

Did you modify the GLDAS metforcing reader to use files such as these?:  GLDAS_NOAH025_3H.A19790201.0000.021.nc4

The current supported GLDAS reader (a bit old now) reads GRIB data.

Can you try this in your lis.config file?   Change/add these lines:

```
Land surface model:                  none
TEMPLATE model timestep:             1hr
```

Then, change your MODEL_OUTPUT file to only output forcing variables.  See if it runs successfully and produces the forcing fields on your output grid.

**Reply by Chanhyuk-Choi** [reply]
Hi @dmocko 

Although GLDAS metforcing was changed to nc4, it was performed without problems in Noah and Noah-MP operation, so the same was applied to Mosaic.
(I haven't dealt with other metforcing data, so I'm putting ERA5's data into the GLDAS settings.)

And I did the mosaic as just mentioned, but I'm getting an error.
[error.txt](https://github.com/NASA-LIS/LISF/files/11333771/error.txt)


with attach lis.config, lis log file and MODEL_OUTPUT file
[lis.config.era5.mosaic.txt](https://github.com/NASA-LIS/LISF/files/11333779/lis.config.era5.mosaic.txt)

[lislog.00.0000.txt](https://github.com/NASA-LIS/LISF/files/11333786/lislog.00.0000.txt)

[TEST_MODEL_OUTPUT_LIST.TBL.txt](https://github.com/NASA-LIS/LISF/files/11333780/TEST_MODEL_OUTPUT_LIST.TBL.txt)

**Reply by dmocko** [reply]
This error indicates the issue is in the "lapse-rate" correction.

Your ldt.config file provides the elevation for your running grid (using SRTM data).  However, it does not provide the GLDAS forcing elevation.  LIS needs both elevations to calculate the difference, to do the lapse-rate correction.

Is there some difference in the ldt.config files between Mosaic and Noah/Noah-MP about the elevation - or in the lis.config files about using a "lapse-rate" correction?

**Reply by Chanhyuk-Choi** [reply]
Thank you for your reply @dmocko 

I'm not sure if I understood completely,

All ldt.config files of Noah/Noah-MP used lapse-rate settings, and lis.config did not use lapse-rate settings.

In the Mosaic lis.config file, I also replaced the lapse-rate setting with "none".

Land surface model: none
TEMPLATE model timestep: 1hr

With the setting, Mosaic operation was successful.

**Reply by dmocko** [reply]
Yes - this is exactly what I meant.  Check your ldt.config files and your lis.config files between the three LSMs.

Glad that my suggestion was able to fix your issue.  Good luck!

**Reply by Chanhyuk-Choi** [reply]
Hello, @dmocko 

informed above

Land surface model: none
TEMPLATE model timestep: 1hr

With the settings, it was confirmed that GLDAS metforcing works successfully,
then change the settings again

Land surface model: Mosaic
Mosaic model timestep: ...

I ran the model with the Mosaic settings, but I'm still getting the same above error.

Maybe it's because I put GLDAS metforcing incorrectly, but I'm in a situation where I can't solve it.

Could you please check it?

[error.txt](https://github.com/NASA-LIS/LISF/files/11350887/error.txt)
[lis.config.era5.mosaic.txt](https://github.com/NASA-LIS/LISF/files/11350894/lis.config.era5.mosaic.txt)

**Reply by dmocko** [reply]
Can you also please send your latest ldt.config for Mosaic, as well as your lislog.0000 file?

I will try to run on my end, although because I don't have your GLDAS forcing code, I will use a different forcing.

**Reply by Chanhyuk-Choi** [reply]
Hi @dmocko 

I'll share the gldas setup I'm using

With the same settings, Noah-MP and Noah run without problems.

Thank you.

[gldaslog.00.0000.txt](https://github.com/NASA-LIS/LISF/files/11363105/gldaslog.00.0000.txt)
[lis.config.gldas.mosaic.txt](https://github.com/NASA-LIS/LISF/files/11363107/lis.config.gldas.mosaic.txt)
[ldt.config.mosaic.0.5grid.gldas.txt](https://github.com/NASA-LIS/LISF/files/11363108/ldt.config.mosaic.0.5grid.gldas.txt)
[2007.zip](https://github.com/NASA-LIS/LISF/files/11363224/2007.zip)
